ingresspipe/schemas/explorer.py

```diff
@@ -439,4 +438,129 @@ def get_digraph_by_edge_type(self, edge_type):
+                        schema_graph_nx.nodes[node]["comment"] = node_to_replace.nodes[replace_node]["comment"]
+                        schema_graph_nx.nodes[node]["description"] = node_to_replace.nodes[replace_node]["description"]
+
+                    if "required" in data and "required" in replace_data:   # replace boolean value of "required" from replacement node
+                        schema_graph_nx.nodes[node]["required"] = node_to_replace.nodes[replace_node]["required"]
+
+                    if "displayName" in data and "displayName" in replace_data: # replace contents of "displayName" from replacement node
+                        schema_graph_nx.nodes[node]["displayName"] = node_to_replace.nodes[replace_node]["displayName"]
+
+                    if "validationRules" in data and "validationRules" in replace_data: # replace contents of "validationRules" from replacement node
+                        schema_graph_nx.nodes[node]["validationRules"] = node_to_replace.nodes[replace_node]["validationRules"]
+
+                    # for the "subClassOf", "requiresDependency", "requiresComponent", "rangeIncludes" fields/keys require rejiggering
+                    # of associated edges
+                    # general strategy we follow for rejiggering is remove edges that existed formerly and add new edges based on contents
+                    # of the replacement node
+
+                    # "subClassOf" key related edge manipulation
+                    if "subClassOf" in data and "subClassOf" in replace_data:
+
+                        # remove formerly existant edges from the master schema/graph
+                        for (u, v) in list(schema_graph_nx.in_edges([node])):
+                            for _edges_to_replace in data["subClassOf"]:
+                                edge_repl = extract_name_from_uri_or_curie(_edges_to_replace["@id"])
+
+                                if edge_repl == u:
+                                    schema_graph_nx.remove_edge(u, v)
+
+                        # extract node names from replacement node and use it to add edges to the master schema/graph
+                        parents = replace_data["subClassOf"]
+                        for _parent in parents:
+                            target_node = extract_name_from_uri_or_curie(_parent["@id"])
+
+                            # label to be associated with "subClassOf" keys is "parentOf"
+                            if target_node != replace_node:
+
+                                # make note of the fact that we are changing in-edges here
+                                schema_graph_nx.add_edge(target_node, replace_node, relationship = "parentOf")
+
+                        # once the edges have been added, change the contents of the node
+                        schema_graph_nx.nodes[node]["subClassOf"] = node_to_replace.nodes[replace_node]["subClassOf"]
```

Comment:
Looks like this code (and variations of it) is repeated where the main difference is the type of edge (e.g. subclassOf, requiresDependency, etc.). @sujaypatil96 do you think it would make sense to abstract as a function with arguments that determine the particular types of edges to be added?

Reply:
Yes, that's exactly right. There's a tonne of redundancy in the code, which I will definitely be removing/abstracting out with a user-defined method. I'm just writing it this way so we can check the behaviour is as expected, and if it works, there are a few more changes I will be making to improve readability, etc.

Reply:
Yes - that works great!